aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -4,10 +4,12 @@
 #include <ATen/Functions.h>
 #include <c10/cuda/CUDACachingAllocator.h>
 #include <c10/cuda/CUDAFunctions.h>
+#include <torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp>
 
 namespace at::cuda {
 
 static bool _cuda_graphs_debug = false;
+constexpr int64_t _busy_wait_millis = 10;
```

Comment:
Could you move the sleeping logic into ProcessGroupNCCL directly? It feels like having a `wait()` there makes sense.
Also side note: 10ms is an eternity for some user, not sure if that's ok?

Reply:
The watchdogs already have sleeping logic, and the fact that wait loop occurs in a side thread (for fault-tolerance reasons) is why we need to wait to resynchronize here. 10ms was chosen here as it is the same polling interval used by the watchdog. I think it's acceptable to pay this as a one-time cost per graph capture---it would be a different story if this was needed for each iteration.

Reply:
Right, this is generic logic to allow the current thread to wait on the watchdog thread to finish. My point is that such logic should live in the distributed code, not the cudagraph code.

Reply:
Do you have a recommendation for how to do this? Right now the function that does the check `watchdogsDone` is in `ProcessGroupNCCL`, but I don't see a clear reason why someone would want to poll this function outside of the graphs use-case.

Reply:
Anyone willing to check if the watchdog is done is most likely going to want to wait on it being done at some point?